guide/popular-topics/errors.md

```diff
@@ -0,0 +1,170 @@
+# Errors
+
+There is no doubt that you have encountered errors while making bots. While errors are very useful at warning you of what is going wrong, many people are stumped by them and how to track them down and fix them, but don't worry, we have you covered. This section will be all about how to diagnose errors, how to identify where they are coming from, and how to fix them.
+
+## Types of Errors
+
+### API Errors
+
+API Errors or DiscordAPIErrors are errors which are thrown by the Discord API when an invalid request is carried out. API Errors can be mostly diagnosed using the message that is given. They can also be further examined by seeing the http method and path used, we will explore tracking these errors down in the next section.
+
+Example: `DiscordAPIError: Cannot send an empty message`
+
+### Discord.js Errors
+
+Discord.js Errors are errors which are thrown by the library itself, they can usually be easily tracked down using the stacktrace and error message.
+
+Example: `Error: Members didn't arrive in time.`
```

Comment:
Just as a note: This error is not easily tracked down since the stacktrace will only point to timers and the library itself, not user code.
This example should probably use a different error (maybe one from RichEmbed/MessageEmbed or failing to resolve role to add / remove, something *sync*).